src/main/scala/viper/gobra/frontend/info/implementation/typing/ghost/GhostIdTyping.scala

```diff
@@ -61,6 +62,11 @@ trait GhostIdTyping { this: TypeInfoImpl =>
 
     case BuiltInFPredicate(tag, _, _) => typ(tag)
     case BuiltInMPredicate(tag, _, _) => typ(tag)
+
+    case ConstructDecl(decl, context) => context.symbType(decl.typ)
+    case DerefDecl(decl, context) => context.symbType(decl.typ)
+    case AssignDecl(decl, context) => context.symbType(decl.typ)
```

Comment:
I moved it to the GhostMiscTyping and defined it in:
```
private[typing] def ghostMemberType(typeMember: GhostTypeMember): Type = typeMember match {
...
    case ConstructDecl(decl, _) => 
      if (decl.isShared) FunctionT(Vector(symbType(decl.typ)), PointerT(symbType(decl.typ)))
      else FunctionT(Vector(symbType(decl.typ)), symbType(decl.typ))
    case DerefDecl(decl, _) => 
      FunctionT(Vector(PointerT(symbType(decl.typ))), symbType(decl.typ))
    case AssignDecl(decl, _) => 
      FunctionT(Vector(PointerT(symbType(decl.typ)), symbType(decl.typ)), VoidType)
}
```
I used this for MemberResolution, so that it can find duplicates.